lib/simple_kafka_consumer/consumer.rb

```diff
@@ -11,15 +12,22 @@ def initialize(kafka_servers, zookeeper_servers, options = {})
         topic_name,
         options
       )
+      Signal.trap("INT") do
+        self.class.terminated = true
+        self.class.timeout = 5
+      end
     end
 
     def run
       debug "partitions: #{consumer.partitions}"
       debug "claimed: #{consumer.claimed}"
       consumer.fetch_loop do |partition, bulk|
-        bulk.each do |message|
-          consume(parse(message))
+        Timeout::timeout(timeout) do
```

Comment:
`Class::class_method` is a deprecated way of calling class methods, please change to `Class.class_method`